### PR TITLE
Tj delete a category

### PR DESCRIPTION
**Pull request by ToniJBaker**
# Description
Trello Ticket #11 Delete a Category
https://trello.com/c/o5EWBEkX/11-11-delete-a-category
1-CategoryRepository lines 98-116 DeleteCategory()
2-ICategoryRepository line 11 DeleteCategory()
3-CategoryController added Delete View for confirmation of deletion, GET and POST the Delete

# Testing Instructions

1. git add and git commit whatever you're working on on your branch
5. git fetch -a
6. git branch -a
7. git checkout tjDeleteCategory
8. If you don't see the most recent version of my files, git pull origin tjDeleteCategory
9. If you haven't created the Tabloid MVC database, please do so.
10. Use this to fill out tables and columns: [SQL data][(https://github.com/NewForce-Cohort-6/tabloidmvc-the-bit-o-honeys/blob/main/SQL/01_TabloidMVC_Create_DB.sql)]
11. Use this to seed the database: [SQL Seed Data][(https://github.com/NewForce-Cohort-6/tabloidmvc-the-bit-o-honeys/blob/main/SQL/02_TabloidMVC_Seed_Data.sql)]
12. Add more seed data for UserProfile: Run SQL Query (adds 1 admin and 3 authors)
```
INSERT INTO UserProfile (DisplayName, FirstName, LastName, Email, CreateDateTime, ImageLocation, UserTypeId)
VALUES ('WebShark', 'Simon', 'Cook', '2manycooks@mail.com', '2022-09-14 13:40:38.926', 'NULL', 1);

INSERT INTO UserProfile (DisplayName, FirstName, LastName, Email, CreateDateTime, ImageLocation, UserTypeId)
VALUES ('Skippy', 'Thomas', 'Green', 'tomgreen@mail.com', '2022-10-26 13:33:38.927', 'NULL', 2);

INSERT INTO UserProfile (DisplayName, FirstName, LastName, Email, CreateDateTime, ImageLocation, UserTypeId)
VALUES ('WingMan', 'John', 'Wiener', 'jwiener33@mail.com', '2022-09-26 13:33:38.927', 'NULL', 2);

INSERT INTO UserProfile (DisplayName, FirstName, LastName, Email, CreateDateTime, ImageLocation, UserTypeId)
VALUES ('SportyGirl', 'Tara', 'Braxton', 'brax06@mail.com', '2022-09-22 13:40:38.927', 'NULL', 2);

SELECT * FROM UserProfile;
```
13. Add more seed data for a Post with different UserProfileId
```
INSERT INTO Post (Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId)
Values ('Growing Artichokes', 'Growing artichokes in your garden is simple and easy to manage. Let us show you how to grow this delicious vegetable at Bonnie Plants.', 'https://cdn.shopify.com/s/files/1/0257/4565/3811/articles/BONNIE-PLANTS_artichokes-iStock-182833592-2400px_d6c556c8-ac94-4d39-be47-8a69ffe552d2.jpg?v=1642541997&width=832', '2021-10-26 13:32:38.927', '2021-10-27 13:33:38.927', 1, 1, 2);

INSERT INTO Post (Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId)
Values ('Fertile Soil', 'Composting reduces pollution, reuses organic waste, reduces the cost of fertilizers and agricultural production inputs and especially returns to soil the nutrients taken to produce food.', 'https://www.fao.org/fileadmin/user_upload/GSP/Areas_of_Work/nutrient_imbalance/z.jpg', '2022-08-26 13:32:38.927', '2022-08-27 13:33:38.927', 1, 1, 2);

INSERT INTO Post (Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId)
Values ('Meet Archaeopteryx, a Feathered Dino With Wings and Teeth', 'It all began with a single feather. On Sept. 30, 1861, the German paleontologist Christian Erich Hermann von Meyer published his description of a jaw-dropping new find. Fossil feathers were unheard of at the time, yet somebody had pulled one out of a limestone quarry near Solnhofen, Bavaria. Von Meyer named the animal it belonged to Archaeopteryx lithographica. His choice was apt; the first part of that name (i.e., Archaeopteryx) means "ancient wing."', 'https://media.hswstatic.com/eyJidWNrZXQiOiJjb250ZW50Lmhzd3N0YXRpYy5jb20iLCJrZXkiOiJnaWZcL2FyY2hhZW9wdGVyeXguanBnIiwiZWRpdHMiOnsicmVzaXplIjp7IndpZHRoIjo4Mjh9LCJ0b0Zvcm1hdCI6ImF2aWYifX0=', '2022-06-26 13:32:38.927', '2022-06-27 13:33:38.927', 1, 1,3); 

INSERT INTO Post (Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId)
Values ('Butterfly Wing Shapes', 'To identify butterflies in this group, look for wings with edges that are jagged, irregular, and angular.', 'https://www.gardenswithwings.com/wp-content/uploads/eacoUOde-450x450.jpg', '2021-06-26 13:32:38.927', '2021-06-27 13:33:38.927', 1, 1, 3);


INSERT INTO Post (Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId)
Values ('Exercises for Women to get Build', 'Strength training for women doesnt have to be complicated to be effective. Heres how to get stronger and see the benefits both in your life and in the mirror!', 'https://www.bodybuilding.com/images/2021/june/these-workouts-are-how-strong-women-get-built-header-830x467.jpg', '2022-03-26 13:32:38.927', '2022-03-27 13:33:38.927', 1, 1, 4);

INSERT INTO Post (Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId)
Values ('Ladies Who Lift', 'In the past few years, weight training for women has increased in popularity. There are several reasons for this, including the rise of social media and common misconceptions about strength training being challenged.', 'https://regymenfitness.com/wp-content/uploads/2021/12/Sportive-serious-people-liftin-980x654.jpg', '2022-03-12 13:32:38.927', '2022-03-15 13:33:38.927', 1, 1, 4);

INSERT INTO Post (Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId)
Values ('9 Guidelines & Best Practices for Exceptional Web Design and Usability', 'When it comes to designing or redesigning a website, it is easy to get hung up on the aesthetics. Does that shade of blue look right? Should the logo be on the right side of the screen, or left?', 'https://img.freepik.com/free-photo/laughing-black-man-glasses-expressing-excitement-emotional-international-student-holding-computer_197531-20167.jpg?w=2000', '2022-02-10 13:32:38.927', '2022-03-02 13:33:38.927', 1, 1, 5);

INSERT INTO Post (Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId)
Values ('Comprehensive Guide to Web Design', 'Web design is a complex subject. Designers and developers have to consider both visual appearance (how the website looks) and functional design (how the website works). With so many things that you need to take into account, the process of design can be tricky. To simplify the task, we’ve prepared this guide for you.', 'https://assets-global.website-files.com/6009ec8cda7f305645c9d91b/60108275cdfc197127f413f9_6002086f72b72727fa01d794_kai-jolly-site.gif', '2022-01-10 13:32:38.927', '2022-01-22 13:33:38.927', 1, 1, 5);
```
14. Click the execute with debugger solid green button to launch the app
15. In browser, please go to Login on the nav bar and log in with this email: admin@example.com
This should redirect you back to the home page.
16. From the nav bar, select the Category Management to view all the categories in alphabetical order.
17. Create a new Category to later Delete.
18. At top of Category list, select New Category.
19. Enter new category name and select "Create" You will be redirected to the Category list and see you new category at the bottom of the list.
20. Directly to the right of the new category you created select the delete button, you will be directed to a confirmation page.
21. Confirm the delete by selecting the red delete button.  You should then go back to the list of categories, minus the category you just deleted.
22. Log out should be showing on the nav bar now. Please click this to log out.
23. You should see a friendly message letting you know you are logged out of the application.
24. Git checkout main or your own branch

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors
- [ ] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

File: TabloidMVC/Repositories/CategoryRepository.cs [hunk 1]
```diff
@@ -95,5 +95,25 @@ OUTPUT INSERTED.ID
...
 
+        public void DeleteCategory(int id)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"
```
Comment (jordan-castelloe) [inline TabloidMVC/Repositories/CategoryRepository.cs:140]: This code works perfectly if the category doesn't have any posts attached to it, but if the category contains one or more posts it throws an exception and reloads the delete form without telling the user what happened. Go find @Tommy-Spurlock and talk to him about options for deleting when you have related data! It's tricky but worth it :)

**Review by jordan-castelloe — CHANGES_REQUESTED** [review]

**Review by TiffersAnn — APPROVED** [review]
Delete confirmation box does not need an "input" section. It has no purpose. But delete functionality works as it should.

**Comment by AllisonCarte** [comment]
Appears to function as it should. Good work!

**Review by TiffersAnn — APPROVED** [review]
Delete works.  Button doesn't show for categories in use.

**Pull request merged**